modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -214,58 +216,32 @@ public ObjectEntry updateObjectEntry(
 
 	@Activate
 	@Modified
-	protected void activate(Map<String, Object> properties) {
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
+
 		_objectConfiguration = ConfigurableUtil.createConfigurable(
 			ObjectConfiguration.class, properties);
-	}
-
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY,
-		target = "(&(com.liferay.object=true)(model.class.name=*))"
-	)
-	protected void setModelResourcePermission(
-		ModelResourcePermission<ObjectEntry> modelResourcePermission,
-		Map<String, Object> properties) {
-
-		String className = (String)properties.get("model.class.name");
-
-		_modelResourcePermissions.put(className, modelResourcePermission);
-	}
-
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		policyOption = ReferencePolicyOption.GREEDY,
-		target = "(&(com.liferay.object=true)(resource.name=*))"
-	)
-	protected void setPortletResourcePermission(
-		PortletResourcePermission portletResourcePermission,
-		Map<String, Object> properties) {
 
-		String resourceName = (String)properties.get("resource.name");
-
-		_portletResourcePermissions.put(
-			resourceName, portletResourcePermission);
+		_modelResourcePermissionsServiceTrackerMap =
+			ServiceTrackerMapFactory.openSingleValueMap(
+				bundleContext,
+				(Class<ModelResourcePermission<ObjectEntry>>)
+					(Class<?>)ModelResourcePermission.class,
+				"(com.liferay.object=true)(model.class.name=*)",
```

Comment:
@W2jx The filter string before is `(&(com.liferay.object=true)(resource.name=*))`, after your change it's `(com.liferay.object=true)(model.class.name=*)`. Are you sure they are equal?

Reply:
I'm sure because I think the call chain is something like this: 
`ServiceTrackerMapFactory.openSingleValueMap -> new ServiceTrackerMapImpl -> ServiceTrackerUtil.createServiceTracker -> filterString =
					"(&" + filterString + "(objectClass=" + clazz.getName() +
						"))"; `